### PR TITLE
Fix the reading of DSA parameters files using the dsaparam app

### PR DESCRIPTION
**Pull request by mattcaswell**
DSA parameters files were failing to load correctly. We also fix a number
of follow on issues which resulted in multiple similar errors messages
being displayed for the same problem, as well as a seg-fault.

### CODE REVIEW

**Review by t8m — APPROVED** [review]
Approved if CI passes.

File: apps/lib/apps.c
Comment (levitte) [inline apps/lib/apps.c:612]: Shouldn't this message say that it's not the expected key type?

**Review by levitte — CHANGES_REQUESTED** [review]

File: apps/lib/apps.c
Comment (mattcaswell) [reply apps/lib/apps.c]: I added some text about this.

**Comment by mattcaswell** [comment]
Fixup pushed, addressing @levitte's comment above.

@t8m - please can you reconfirm your approval?

File: apps/lib/apps.c
Comment (t8m) [inline apps/lib/apps.c:613]: Typo: paraeters :grin:

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by mattcaswell** [comment]
Grrr. Typo fixed.

**Review by t8m — APPROVED** [review]

File: apps/lib/apps.c
Comment (levitte) [inline apps/lib/apps.c:618]: You lost the error printout for `params == NULL`

(I'm not sure why we still have those error printouts, since `load_key_certs_crls()` does that already, but removing it only here makes this function inconsistent with `load_key` and `load_pubkey`)

Comment (mattcaswell) [reply apps/lib/apps.c:618]: This change was deliberate. As per my introductory comments:

"We also fix a number of follow on issues which resulted in multiple similar errors messages being displayed for the same problem,"

>  but removing it only here makes this function inconsistent with load_key and load_pubkey

That may be true, but I consider those functions out of scope of this change.

Comment (levitte) [reply apps/lib/apps.c:618]: So, will that be fixed at some point?

Comment (mattcaswell) [reply apps/lib/apps.c:618]: I'll do it in a separate PR. I don't want to hold up merging this one.

Comment (mattcaswell) [reply apps/lib/apps.c:618]: #13355

**Review by slontis — COMMENTED** [review]
LGTM.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**
